MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -3077,11 +3077,24 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 		pWritableExtns->vk_KHR_fragment_shader_barycentric.enabled = false;
 		pWritableExtns->vk_NV_fragment_shader_barycentric.enabled = false;
 	}
-	// gpuAddress requires Tier2 argument buffer support (per feedback from Apple engineers).
-	if (_metalFeatures.argumentBuffersTier < MTLArgumentBuffersTier2) {
+    
+    // The relevant functions are not available if not built with Xcode 14.
+#if !MVK_XCODE_14
+    pWritableExtns->vk_KHR_buffer_device_address.enabled = false;
+    pWritableExtns->vk_EXT_buffer_device_address.enabled = false;
+#else
+	// Xcode warns about invalid use if we use !@available, so instead this abomination.
+    if (@available(macOS 13.0, iOS 16.0, *)) {} else {
```

Comment:
Are we sure we need this check? Isn't it redundant to the limits on OS versions set below in `MVKExtensions.def`?

If we do need this check, try the following MoltenVK function instead. It is standard in MoltenVK (as a C++ library, we try to minimize Apple artifacts wherever possible), and it can be logically negated.

https://github.com/KhronosGroup/MoltenVK/blob/7740554a9c2a7f64c9f8cee620569e394ff5f7e6/Common/MVKOSExtensions.h#L55

Reply:
Yeah, true. I just personally find the MVKExtension.* files quite complicated so I find it hard to read, but yes, the check is in there.